docs/docs/render-callbacks.md

```diff
@@ -0,0 +1,56 @@
+---
+id: render-callbacks
+title: Render Callbacks
+permalink: docs/render-callbacks.html
+---
+
+Render callbacks are an advanced technique for certain types of asynchronous rendering. A render callback is a function that is passed as a prop to a React element, and is called at some point in the future. The render callback is typically passed additional data that isn't available at the time the element is rendered:
+
+```js
+render() {
+  return (
+    <FetchData
+      url="/path/to/data"
+      renderOnSuccess={(data) => <ChildComponent data={data} />}
+    />
+  );
+}
+```
+
+In this example, the `FetchData` component fetches some data from an endpoint. While the data is loading, it renders nothing. Once the data is ready, the render callback `renderOnSuccess` is used to compute the rendered content. `FetchData` could also be implemented to accept an additional render callback that specifies the content to render while loading, and another that handles an error.
+
+Render callbacks work by giving a child component control over how and when to render part of the UI.
```

Comment:
Ah yes, I meant compared to passing the UI as a child.

```js
<Child><Foo /></Child>
```

versus

```js
// child controls when to render Foo
<Child>{() => <Foo />}</Child>
```

But I see what you mean. I'll rewrite.